Stripe/STPPaymentContext.m

```diff
@@ -325,18 +377,26 @@ - (void)pushShippingViewController {
     WEAK(self);
     [self.didAppearPromise voidOnSuccess:^{
         STRONG(self);
-        STPShippingAddressViewController *addressViewController = [[STPShippingAddressViewController alloc] initWithPaymentContext:self];
-        [navigationController pushViewController:addressViewController animated:YES];
+        if (self.state == STPPaymentContextStateNone) {
+            self.state = STPPaymentContextStateShowingRequestedViewController;
+
+            STPShippingAddressViewController *addressViewController = [[STPShippingAddressViewController alloc] initWithPaymentContext:self];
+            [navigationController pushViewController:addressViewController animated:YES];
+        }
+        else {
```

Comment:
is this else block correct? (seems like we should do nothing, rather than resetting state)